src/components/renderer/field/form-field-renderer.component.tsx

```diff
@@ -172,6 +172,18 @@ export const FormFieldRenderer = ({ fieldId, valueAdapter, repeatOptions }: Form
       />
     );
   }
+
+  // If the field is transient and has no value, we do not render it in embedded view mode.
+  // This is to prevent transient fields from being displayed in the form when they are not necessarily needed.
+  if (
+    sessionMode === 'embedded-view' &&
+    field.questionOptions.isTransient &&
```

Comment:
@samuelmale do you if the fields are automatically marked as transient when they have no value? Am having second thoughts about the `&&` on both ends of this condition

Reply:
Well, the form-engine ignores all transient fields on submission, so such fields don't track persistent values. Check out this Slack [thread](https://openmrs.slack.com/archives/CHP5QAE5R/p1753362291489839?thread_ts=1753105429.463849&cid=CHP5QAE5R) for more context.